util/energy_saving_calculation.py

```diff
@@ -0,0 +1,228 @@
+from datetime import datetime, time, timedelta
+import json
+import os
+import logging
+from dateutil import parser
+import redis
+import pytz
+from models.user import User
+from util.error import exception_message
+from util.database import create_session, get_engine
+from util.redis_helpers import get_sorted_keys
+
+
+logging.basicConfig()
+logger = logging.getLogger('util/energy_saving_calculation')
+redis_host = os.environ['REDIS_HOST']
+redis_port = os.environ['REDIS_PORT']
+redis_db = os.environ['REDIS_DB']
+redis_client = redis.Redis(host=redis_host, port=redis_port, db=redis_db)
+
+
+def get_all_user_meter_ids(session):
+    """ Get all user meter ids from sqlite database. """
+
+    return [meter_id[0] for meter_id in session.query(User.meter_id).all()]
+
+
+def calc_ratio_values(start):
+    """ Calculates the percentages of energy consumption for the specified
+    term. A term is a year where the start may be specified by the caller.
+    :param datetime.date start: the start date of the term
+    :return: sum of all standard load profile ratio values of the given term
+    :rtype: float
+    """
+
+    end = datetime(start.year + 1, start.month, start.day).date()
+    term_end = datetime.utcnow().date()
+    engine = get_engine()
+    energy_total = 0.0
+    ratio_values = 0.0
+    try:
+        with engine.connect() as con:
+            rs = con.execute("SELECT * FROM loadprofile WHERE date BETWEEN \'" +
+                             str(start) + "\' AND \'" + str(end) + '\' ORDER BY date')
+
+            # Calculate energy total which should be ~1.000.000 kWh
+            for row in rs:
+                energy_total += row[2]
+
+            # Calculate percentage of each energy value and write to result
+            rs = con.execute("SELECT * FROM loadprofile WHERE date BETWEEN \'" +
+                             str(start) + "\' AND \'" + str(term_end) + '\' ORDER BY date')
+            for row in rs:
+                percentage = float(row[2])/energy_total
```

Comment:
If this is still needed consider another name for percentage.. something like `partial_of_ongoing_year`... maybe shorter ;)